src/Components/Navbar/Navbar.jsx

```diff
@@ -1,6 +1,121 @@
+import { useState, useEffect, useRef } from "react";
+import { NavLink } from "react-router-dom";
 import style from "./Navbar.module.scss";
+
 const Navbar = () => {
-  return <nav className={style.navbar}>Navbar</nav>;
+  const [burger, setBurger] = useState(false);
+
+  const toggleMenu = () => {
+    setBurger(!burger);
+  };
+  const handleKeyDown = (e) => {
+    if (e.key === "Enter") {
+      toggleMenu();
+    }
+  };
+
+  const menuRef = useRef();
+  const hamRef = useRef();
+
+  useEffect(() => {
+    const handler = (e) => {
+      if (!menuRef.current.contains(e.target) && !hamRef.current.contains(e.target)) {
+        setBurger(false);
+      }
+    };
+    document.addEventListener("mousedown", handler);
+    return () => {
+      document.removeEventListener("mousedown", handler);
+    };
+  });
+
+  useEffect(() => {
+    if (burger === true) {
+      document.body.style.height = "100dvh";
+      document.body.style.overflow = "hidden";
+    } else {
+      document.body.style.height = "";
+      document.body.style.overflow = "";
+    }
+  });
```

Comment:
never use useEffect hook without a dependency array as it can lead to unintended execution of code. 

Additionally, can you explain why are you updating body styles ?

Reply:
To fix the body and disable scrolling when the navmenu is active

Reply:
OK got it, but provide proper useEffect dependencies.